core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/profile/DefaultProfileConfigurationManager.java

```diff
@@ -39,21 +39,31 @@ public DefaultProfileConfigurationManager(Map<String, ProfileConfiguration> prof
     }
 
     public ProfileConfiguration getProfileConfig(String profileName) throws UserStoreException {
-        return profileConfigs.get(profileName);
+        if (profileName != null) {
+            return profileConfigs.get(profileName);
+        } else {
+            return null;
```

Comment:
Cannot we avoid returning null here, as it is the root cause for NPEs and adding the complexity of checking for null results by caller functions ?